_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+author: [deepu-sasidharan, chase-aucoin]
+by: advocate
+communities: [.net]
+description: "This is a tutorial on how to secure your .NET containerized chat app with AWS ECS Fargate and Okta."
+tags: [containers, aws, fargate, c-sharp, csharp, dotnet, dot-net, asp-dot-net, vuejs]
+tweets:
+  - "Ever played with AWS Fargate? Learn how to authenticate users to your #dotnetcore #container hosted in #aws ->."
+  - "Unsure about #containerized #dotnet with #Fargate in #AWS? We've got a primer just waiting for you ->"
+  - "Learn how to create secure .NET #containers in #AWS with Fargate ->"
+image: blog/featured/okta-dotnet-mouse-down.jpg
+type: conversion
+---
+
+In a previous article, we learned [how to host a serverless .NET application using AWS Lambda](/blog/2020/06/08/serverless-lambda-functions-csharp). We talked about the history of serverless and how companies are using these types of technology to simplify delivering APIs and functionality faster than traditional methods. Some problems will arise in this type of application when you need more capability than your standard HTTP requests like GET, POST, PUT, DELETE, etc. Web Sockets is a great example of this.
+
+{% include toc.md %}
+
+## Understanding Containers vs. Virtual Machines
+
+Around 2013, [Docker](https://www.docker.com/) was released, beginning a shift in how we think about hosting applications and managing infrastructure. Infrastructure teams and operations have been leveraging Virtual Machines with great success for decades, so you may be tempted to think, "isn't this just a virtual machine?" In spirit, yes; in application, no.
+
+To talk about containers, we have to talk about Virtual Machines. A virtual machine (VM) is a virtual representation of a computer all the way from its boot process to loading an entire operating system. This provides a lot of flexibility because you gain fine-grained control over an entire virtual computer, but it brings all the pitfalls of a whole computer as well. The operating system has to be maintained, patched, and updated; disks have to be managed, and all manner of operational overhead goes into their care and feeding. Notably, VMs are also huge because an entire OS has to go on them. So when you start a VM, you have to go through the entire boot sequence of a traditional computer.
+
+These are the three problem areas that containers hope to solve; Ease of management, size, and speed to start. There are others, but for simplicity, we will focus on those three as I feel (IMHO) they are the most relevant to business processes. Containers differ from a traditional virtual machine in one significant way. They do not have an actual Operating System. This may come as a surprise to you, even if you might have dabbled with containers before. Containers instead have an abstraction of an operating system that provides hooks into the host operating system using standards and conventions supplied by the container system.
```

Comment:
```suggestion
There are three major problem areas that containers hope to solve: ease of management, size, and speed to start. There are others, but for simplicity, I will focus on those that are the most relevant to business processes. Containers differ from traditional virtual machines in one significant way. They do not have an actual OS. This may come as a surprise to you, even if you've dabbled in containers before. Instead, containers rely on an abstraction of an operating system that provides hooks into the host OS using standards and conventions supplied by the container system.
```
conciseness, clarity, alignment of singular/plural noun usage in a single sentence